src/utilities/layer-from-json.ts

```diff
@@ -0,0 +1,22 @@
+import * as layer from '../layer';
+import { get } from 'lodash';
+import { ILayer, ILayerJSON } from '../layer/base-layer';
+
+export default function layerFromJSON(jsonLayer: ILayerJSON, inputLayer1?: ILayer) { 
+  if (!layer.hasOwnProperty(jsonLayer.type)) return null;
+
+  const Layer: Function = get(layer, `${layer}.${jsonLayer.type}`);
```

Comment:
`Function` should be replaced by proper `Layer` interface/type. Instead of get we have to use own method to get later. For more information please see the conversation on here: https://github.com/BrainJS/brain.js/pull/611/files

Reply:
Reflect.construct() does not accept anything else than Function :)